### PR TITLE
Porting CC 043 Context Free Grammar Generator

### PR DESCRIPTION
**Pull request by anusha-exe**
🔨 Added: 

-  timestamps
-  links to code example, references, tutorial playlist, livestream archive
-  descriptions, icons, tags
-  thumbnails

Ready for [preview](https://deploy-preview-713--codingtrain.netlify.app/challenges/3) 🚂 

## 📋  Notes:

1. This challenge had no generated JSON so I have added what I found from the YouTube videos and whatever felt right to me. So, the description and the links needs to be checked and if anything is missing or unnecessary, please tell me.
2.  From what I saw, this challenge had no showcase contributions either. 
~3.  It is an old challenge so there isn't any web editor p5.js version so I have kept the link to the repo in the `p5:` field. If you can create a web editor version, then I'll update the links~
4.  I added new words to the text generator. I hope this thumbnails are okay. ~Perhaps the font size is too small and unreadable? And the original thumbnail from `images` is also not legible it seems~
5.  This CC is a part of the playlist on Context-Free-Grammar. That playlist is a prospective sidetrack. If no one has started working on it, I could do it

### CODE REVIEW

**Comment by dipamsen** [comment]
@anusha-exe  Thanks for your efforts! Here is the p5.js web editor sketch: https://editor.p5js.org/codingtrain/sketches/_ll_PYXeg

(If the json is not available, the p5.js web editor sketch is usually found on the old website.)

**Comment by anusha-exe** [comment]
@dipamsen Oh! Thanks. I was only looking at the GitHub archive. I'll make the changes

**Comment by shiffman** [comment]
Wonderful work @anusha-exe!! I agree, I think the thumbnail might look better with even just one example phrase but much larger. That said it is terrific as is! Let me know if you'd like me to go ahead and merge (and we can always consider updating the thumb another time) or if you want to make any more changes!

**Comment by kfahn22** [comment]
Thanks, another great job!  Another option for relatedChallenges is  "39-madlibs-generator" if you wanted to add.

**Comment by anusha-exe** [comment]
@kfahn22 My bad, I forgot to paste the link there. Thank you for the corrections!
@shiffman I have updated the thumb. I think it can be merged now!

File: content/videos/challenges/43-context-free-grammar/index.json [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    "date": "2016-10-31",
+    "languages": ["p5.js", "JavaScript"],
+    "topics": ["text-generator", "recursion", "grammar"],
+    "canContribute": true,
+    "relatedChallenges": ["16-l-system-fractal-trees", "14-fractal-trees-recursive", "77-recursion", "39-madlibs-generator"],
+    "timestamps": [
+      { "time": "0:00", "title": "Introduction!" },
+      { "time": "1:10", "title": "Defining a JavaScript object containing the grammar rules" },
+      { "time": "3:44", "title": "Creating a recursive algorithm to generate a random sentence" },
+      { "time": "11:51", "title": "Generating multiple random suntences" },
```
Comment (dipamsen) [inline content/videos/challenges/43-context-free-grammar/index.json:15]:
```suggestion
      { "time": "11:51", "title": "Generating multiple random sentences" },

```

Comment (anusha-exe) [reply content/videos/challenges/43-context-free-grammar/index.json]: Thanks a lot. I'll double check them next time

**Comment by dipamsen** [comment]
I think `Live Stream Archive` as the title is good when there is only one link, the specification is only required if multiple links are present.

**Comment by kfahn22** [comment]
My apologies, the link to the Live Stream archive was a recent addition.  We can certainly update the guide.  Thank you for all of the work you have done!!!

**Comment by shiffman** [comment]
looks fantastic!!

**Comment by anusha-exe** [comment]
@kfahn22 please, it's nothing to apologise for! I'd be happy to help in any way

**Pull request merged**